tests/integration/test_charm.py

```diff
@@ -301,8 +352,22 @@ async def test_integration_with_zookeeper(ops_test: OpsTest, charm_versions):
     await ops_test.model.wait_for_idle(
         apps=[APP_NAME, charm_versions.zookeeper.application_name], timeout=1000, status="active"
     )
+    kyuubi_leader = await find_leader_unit(ops_test, app_name=APP_NAME)
+    assert kyuubi_leader is not None
+
+    logger.info("Running action 'get-password' on kyuubi-k8s unit...")
```

Comment:
One thing that I don't like here is the repetition of these lines across all test functions. However, we can defer that for now, since we want to have this merged ASAP. We can clean this later in a different PR, but let's create a marker/ticket in the backlog so we don't miss it later.

Reply:
Yeah I'll streamline all of these with jubilant

Reply:
:tada: